### ISSUE TITLE
Bug in Federated Learning on MNIST using a CNN Example

### ISSUE THREAD

**Issue by RohitAg13** [OpenMined/PySyft#2267]
**Describe the bug**
Error on FederatedDataLoader step while running [Federated Learning on MNIST using a CNN](https://github.com/OpenMined/PySyft/blob/master/examples/tutorials/Part%206%20-%20Federated%20Learning%20on%20MNIST%20using%20a%20CNN.ipynb)

**To Reproduce**
![image](https://user-images.githubusercontent.com/17871166/59260179-ca445f80-8c58-11e9-81b6-9fd32f866a51.png)

**Error**
---------------------------------------------------------------------------
```
KeyError                                  Traceback (most recent call last)
~/anaconda3/envs/pysyft/lib/python3.7/site-packages/syft/frameworks/torch/hook/hook_args.py in hook_function_args(attr, args, kwargs, return_args_type)
    144         # TODO rename registry or use another one than for methods
--> 145         hook_args = hook_method_args_functions[attr]
    146         get_tensor_type_function = get_tensor_type_functions[attr]

KeyError: 'torch.as_tensor'

During handling of the above exception, another exception occurred:

IndexError                                Traceback (most recent call last)
<ipython-input-5-4085cd6569bc> in <module>
      5                        transforms.Normalize((0.1307,), (0.3081,))
      6                    ]))
----> 7     .federate((bob, alice)), # <-- NEW: we distribute the dataset across all the workers, it's now a FederatedDataset
      8     batch_size=args.batch_size, shuffle=True, **kwargs)
      9 

~/anaconda3/envs/pysyft/lib/python3.7/site-packages/syft/frameworks/torch/federated/dataset.py in dataset_federate(dataset, workers)
    159     datasets = []
    160     data_loader = torch.utils.data.DataLoader(dataset, batch_size=data_size)
--> 161     for dataset_idx, (data, targets) in enumerate(data_loader):
    162         worker = workers[dataset_idx % len(workers)]
    163         logger.debug("Sending data to worker %s", worker.id)

~/anaconda3/envs/pysyft/lib/python3.7/site-packages/torch/utils/data/dataloader.py in __next__(self)
    558         if self.num_workers == 0:  # same-process loading
    559             indices = next(self.sample_iter)  # may raise StopIteration
--> 560             batch = self.collate_fn([self.dataset[i] for i in indices])
    561             if self.pin_memory:
    562                 batch = _utils.pin_memory.pin_memory_batch(batch)

~/anaconda3/envs/pysyft/lib/python3.7/site-packages/torch/utils/data/dataloader.py in <listcomp>(.0)
    558         if self.num_workers == 0:  # same-process loading
    559             indices = next(self.sample_iter)  # may raise StopIteration
--> 560             batch = self.collate_fn([self.dataset[i] for i in indices])
    561             if self.pin_memory:
    562                 batch = _utils.pin_memory.pin_memory_batch(batch)

~/anaconda3/envs/pysyft/lib/python3.7/site-packages/torchvision/datasets/mnist.py in __getitem__(self, index)
     93 
     94         if self.transform is not None:
---> 95             img = self.transform(img)
     96 
     97         if self.target_transform is not None:

~/anaconda3/envs/pysyft/lib/python3.7/site-packages/torchvision/transforms/transforms.py in __call__(self, img)
     59     def __call__(self, img):
     60         for t in self.transforms:
---> 61             img = t(img)
     62         return img
     63 

~/anaconda3/envs/pysyft/lib/python3.7/site-packages/torchvision/transforms/transforms.py in __call__(self, tensor)
    162             Tensor: Normalized Tensor image.
    163         """
--> 164         return F.normalize(tensor, self.mean, self.std, self.inplace)
    165 
    166     def __repr__(self):

~/anaconda3/envs/pysyft/lib/python3.7/site-packages/torchvision/transforms/functional.py in normalize(tensor, mean, std, inplace)
    204         tensor = tensor.clone()
    205 
--> 206     mean = torch.as_tensor(mean, dtype=torch.float32, device=tensor.device)
    207     std = torch.as_tensor(std, dtype=torch.float32, device=tensor.device)
    208     tensor.sub_(mean[:, None, None]).div_(std[:, None, None])

~/anaconda3/envs/pysyft/lib/python3.7/site-packages/syft/frameworks/torch/hook/hook.py in overloaded_func(*args, **kwargs)
    705             cmd_name = f"{attr.__module__}.{attr.__name__}"
    706             command = (cmd_name, None, args, kwargs)
--> 707             response = TorchTensor.handle_func_command(command)
    708             return response
    709 

~/anaconda3/envs/pysyft/lib/python3.7/site-packages/syft/frameworks/torch/tensors/interpreters/native.py in handle_func_command(cls, command)
    197             # Note that we return also args_type which helps handling case 3 in the docstring
    198             new_args, new_kwargs, new_type, args_type = syft.frameworks.torch.hook_args.hook_function_args(
--> 199                 cmd, args, kwargs, return_args_type=True
    200             )
    201             # This handles case 3: it redirects the command to the appropriate class depending

~/anaconda3/envs/pysyft/lib/python3.7/site-packages/syft/frameworks/torch/hook/hook_args.py in hook_function_args(attr, args, kwargs, return_args_type)
    150     except (IndexError, KeyError, AssertionError):  # Update the function in case of an error
    151         args_hook_function, get_tensor_type_function = build_hook_args_function(
--> 152             args, return_tuple=True
    153         )
    154         # Store the utility functions in registries

~/anaconda3/envs/pysyft/lib/python3.7/site-packages/syft/frameworks/torch/hook/hook_args.py in build_hook_args_function(args, return_tuple)
    180     # Build a function with this rule to efficiently the child type of the
    181     # tensor found in the args
--> 182     get_tensor_type_function = build_get_tensor_type(rule)
    183     return args_hook_function, get_tensor_type_function
    184 

~/anaconda3/envs/pysyft/lib/python3.7/site-packages/syft/frameworks/torch/hook/hook_args.py in build_get_tensor_type(rules, layer)
    409 
    410     if first_layer:
--> 411         return lambdas[0]
    412     else:
    413         return lambdas

IndexError: list index out of range
```

**Desktop:**
 - OS: Ubuntu 16.04
 - Python: 3.7.3
 - torch : 1.1.0
 - torchvision: 0.3.0 
 - syft: 0.1.16a1
 - No GPU or Cuda

**Comment by RohitAg13** [comment]
Issue is solved by moving to `torch==1.0.1` and `torchvision==0.2.2`

**Comment by LaRiffle** [comment]
Will be solved in syft>=0.1.18 (coming soon)

**Comment by RohitAg13** [comment]
Thanks
